keras_cv/models/resnet.py

```diff
@@ -0,0 +1,537 @@
+        (True for ResNetV2, False for ResNet).
+      include_rescaling: whether or not to Rescale the inputs.If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+      name: string, model name.
+      include_top: whether to include the fully-connected
+        layer at the top of the network.
+      weights: one of `None` (random initialization),
+        or the path to the weights file to be loaded.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      num_classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      **kwargs: For backwards compatibility only.
+
+        Returns:
+      A `keras.Model` instance.
+    """
+    if kwargs:
+        raise ValueError("Unknown argument(s): %s" % (kwargs,))
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either `None` or the path to the "
+            "weights file to be loaded. Weights file not found at location: {weights}"
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    img_input = layers.Input(shape=input_shape)
+
+    if include_rescaling:
+        img_input = layers.Rescaling(1 / 255.0)(img_input)
+
+    x = layers.Conv2D(
+        64, 7, strides=2, use_bias=True, padding="same", name="conv1_conv"
+    )(img_input)
+
+    if not preact:
+        x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="conv1_bn")(
+            x
+        )
+        x = layers.Activation("relu", name="conv1_relu")(x)
+
+    x = layers.MaxPooling2D(3, strides=2, padding="same", name="pool1_pool")(x)
+
+    x = stack_fn(x)
+
+    if preact:
+        x = layers.BatchNormalization(axis=BN_AXIS, epsilon=1.001e-5, name="post_bn")(x)
+        x = layers.Activation("relu", name="post_relu")(x)
```

Comment:
Would it make sense to add a comment here denoting the differences between ResNet v1 and v2?

Reply:
Nevermind, I see that in the docstring.

Reply:
a comment would be helpful too.  I agree - the resnet code is harder to follow due to the reuse